dbms/src/Storages/Page/V3/spacemap/SpaceMap.h

```diff
@@ -95,9 +95,11 @@ class SpaceMap
     virtual std::tuple<UInt64, UInt64, bool> searchInsertOffset(size_t size) = 0;
 
     /**
-     * Get the offset of the last free block. `[margin_offset, +∞)` is not used at all.
+     * Get the used boundary of current SpaceMap. 
+     * If last node is [xxx, end]. Then `[margin_offset, +∞)` is not used at all. Return the `start` of last node.
+     * If last node is not [xxx, end]. Then right margin must be `end`.
```

Comment:
```suggestion
     * Get the used boundary of this SpaceMap. 
     * The return value (`used_boundary`) means that `[used_bounary + 1, +∞)` is safe to be truncated.
     * If the `used_boundary` is equal to the `end` of this SpaceMap, it means that there is no space to be truncated.
```

Reply:
The interface should not describe the details of implementation. Here the "last node" is implementation details.
It should describe what the return value is and whether there is a special return value.